packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -114,19 +115,35 @@ export const useArbTokenBridge = (
 
   const { addPendingTransaction } = useTransactionHistory(walletAddress)
 
+  const { destinationAddress } = useDestinationAddressStore()
+
   const {
     eth: [, updateEthL1Balance],
     erc20: [, updateErc20L1Balance]
   } = useBalance({
     provider: l1.provider,
-    walletAddress
+    walletAddress: walletAddress
   })
   const {
     eth: [, updateEthL2Balance],
     erc20: [, updateErc20L2Balance]
   } = useBalance({
     provider: l2.provider,
-    walletAddress
+    walletAddress: walletAddress
+  })
+  const {
+    eth: [, updateEthL1CustomDestinationBalance],
+    erc20: [, updateErc20L1CustomDestinationBalance]
+  } = useBalance({
+    provider: l1.provider,
+    walletAddress: destinationAddress
+  })
+  const {
+    eth: [, updateEthL2CustomDestinationBalance],
+    erc20: [, updateErc20CustomDestinationL2Balance]
+  } = useBalance({
+    provider: l2.provider,
+    walletAddress: destinationAddress
   })
```

Comment:
ok this works because we are forcefully trying to update everything, including making unnecessary updates

but we must be aware of it and refactor all these `useBalance` hooks to using `source` & `destination`-based args instead of `l1` and `l2` args

please file a followup ticket as this change is introducing technical debt to the code base

Reply:
Created a ticket to update useBalance to use source-destination based nomenclature in general. https://linear.app/offchain-labs/issue/FS-329/refactor-usebalance-hook-to-use-source-destination-chain-based-naming

But regarding this redundant code in question, we won't be needing this after Bridge-SDK changes come into effect (which is after reskin) because we're getting rid of balance updation / transfers from `useArbTokenBridge`. Everything will be handled in TransferPanel and there we have access to `isDepositMode` and balance updation handling is much better there.